cmake/tpl/FindKLU.cmake

```
@@ -30,6 +30,18 @@
 #   KLU_LIBRARIES   - all of the libraries needed for KLU
 # ---------------------------------------------------------------
 
+if (NOT (KLU_INCLUDE_DIR OR KLU_LIBRARY_DIR OR KLU_LIBRARY))
+  # prefer import target from upstream SuiteSparse if available and
+  # user didn't point to a specific (different) version
+  include ( CMakeFindDependencyMacro )
+  find_dependency(KLU CONFIG)
```

```suggestion
  find_package(KLU REQUIRED)
```
We prefer to use `find_package`.

---

I can switch to `find_dependency` (but that won't propagate flags like `SILENT` from the parent).
Adding `REQUIRED` would break the configuration if the installed SuiteSparse is older than version 7.4.0.
Removing `CONFIG` might not fetch the `KLUConfig.cmake` as installed by SuiteSparse 7.4.0 or later.

---

OK, we can remove `REQUIRED` to allow for the older versions, but we will need to use `find_package` for consistency with how we treat other TPLs (we use `find_dependency` in the SUNDIALSConfig.cmake, but not in our custom Find modules).